specification/under_development/Compilation_And_Targeting.md

```diff
@@ -0,0 +1,67 @@
+# Compilation Process and Toolchain
+
+QIR serves as an integrating layer on top of which a shared compilation stage is built to connect a variety of front- and backends. To accelerate progress, quantum programming frameworks often choose to expose tools and APIs to facilitate reseach and development efforts across the entire stack rather than just at the application level. Efforts to develop and evaluate the performance of different error correction schemes, for example, illustrate that compilation logic may initially be expressed as application code before it is automated and integrated into the compiler toolchain. 
```

Comment:
Probably worth defining somewhere what is meant by "frontend" and "backend" as these are critical terms that are used a lot.

Reply:
Agreed, but also always hard to do. My current plan is to a) link to a wiki page that describes compiler front-, middle, and backends, which seems accurate enough to apply, and b) add a diagram with a caption somewhere to maybe visualize the compilation stages and add a sentence or two there. I'll omit services from the mix - that would like just add confusion otherwise.

Reply:
I added the links and added a todo for the diagram. I'll close this thread for now, but please comment if you feel it is not sufficiently addressed.